clang/clangArgumentParser.go

```diff
@@ -82,15 +84,23 @@ func EvaluatePreprocessedFile(buildRoot string, command *CompilerCommand) ([]byt
 	}
 
 	// read the contents of the file am hash it
-	f, err := os.Open(filename)
-	if err != nil {
-		return nil, err
-	}
-	defer f.Close()
-
 	hasher := sha256.New()
-	if _, err := io.Copy(hasher, f); err != nil {
-		return nil, err
+	if len(baseDir) == 0 {
+		f, err := os.Open(filename)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
```

Comment:
This causes a bug that is fixed in #17. The file is not closed before calling `os.Remove` at line 110, which results in an error.